File: src/main/java/org/example/model/Consultation.java
package org.example.model;

import java.time.LocalDateTime;

public class Consultation {

    private int id;
    private int patientId;
    private String patientNom;
    private String patientEmail;
    private Integer pathologieId;
    private String pathologieNom;
    private LocalDateTime dateConsultation;
    private String motif;
    private String diagnostic;
    private String observations;
    private String ordonnance;
    private String statut;

    public Consultation() {}

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getPatientId() { return patientId; }
    public void setPatientId(int patientId) { this.patientId = patientId; }

    public String getPatientNom() { return patientNom; }
    public void setPatientNom(String patientNom) { this.patientNom = patientNom; }

    public String getPatientEmail() { return patientEmail; }
    public void setPatientEmail(String patientEmail) { this.patientEmail = patientEmail; }

    public Integer getPathologieId() { return pathologieId; }
    public void setPathologieId(Integer pathologieId) { this.pathologieId = pathologieId; }

    public String getPathologieNom() { return pathologieNom; }
    public void setPathologieNom(String pathologieNom) { this.pathologieNom = pathologieNom; }

    public LocalDateTime getDateConsultation() { return dateConsultation; }
    public void setDateConsultation(LocalDateTime dateConsultation) { this.dateConsultation = dateConsultation; }

    public String getMotif() { return motif; }
    public void setMotif(String motif) { this.motif = motif; }

    public String getDiagnostic() { return diagnostic; }
    public void setDiagnostic(String diagnostic) { this.diagnostic = diagnostic; }

    public String getObservations() { return observations; }
    public void setObservations(String observations) { this.observations = observations; }

    public String getOrdonnance() { return ordonnance; }
    public void setOrdonnance(String ordonnance) { this.ordonnance = ordonnance; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    @Override
    public String toString() {
        return "Consultation #" + id + " - " + motif + " (" + statut + ")";
    }
}

File: symfony-integ/src/Entity/Consultation.php
<?php
namespace App\Entity;
use App\Repository\ConsultationRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ConsultationRepository::class)]
#[ORM\Table(name: 'consultation')]
class Consultation
{
    #[ORM\Id] #[ORM\GeneratedValue] #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name: 'patient_id')]
    private ?int $patientId = null;

    #[ORM\Column(name: 'patologie_id', nullable: true)]
    private ?int $patologieId = null;

    #[ORM\Column(name: 'date_consultation', type: 'datetime')]
    private ?\DateTimeInterface $dateConsultation = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $motif = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $diagnostic = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $observations = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $ordonnance = null;

    #[ORM\Column(length: 50, options: ['default' => 'planifiée'])]
    private string $statut = 'planifiée';

    // Non-mapped
    private ?string $patientNom = null;
    private ?string $patientEmail = null;
    private ?string $patologieNom = null;

    public function getId(): ?int { return $this->id; }
    public function getPatientId(): ?int { return $this->patientId; }
    public function setPatientId(int $v): static { $this->patientId = $v; return $this; }
    public function getPatologieId(): ?int { return $this->patologieId; }
    public function setPatologieId(?int $v): static { $this->patologieId = $v; return $this; }
    public function getDateConsultation(): ?\DateTimeInterface { return $this->dateConsultation; }
    public function setDateConsultation(?\DateTimeInterface $v): static { $this->dateConsultation = $v; return $this; }
    public function getMotif(): ?string { return $this->motif; }
    public function setMotif(?string $v): static { $this->motif = $v; return $this; }
    public function getDiagnostic(): ?string { return $this->diagnostic; }
    public function setDiagnostic(?string $v): static { $this->diagnostic = $v; return $this; }
    public function getObservations(): ?string { return $this->observations; }
    public function setObservations(?string $v): static { $this->observations = $v; return $this; }
    public function getOrdonnance(): ?string { return $this->ordonnance; }
    public function setOrdonnance(?string $v): static { $this->ordonnance = $v; return $this; }
    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $v): static { $this->statut = $v; return $this; }
    public function getPatientNom(): ?string { return $this->patientNom; }
    public function setPatientNom(?string $v): static { $this->patientNom = $v; return $this; }
    public function getPatientEmail(): ?string { return $this->patientEmail; }
    public function setPatientEmail(?string $v): static { $this->patientEmail = $v; return $this; }
    public function getPatologieNom(): ?string { return $this->patologieNom; }
    public function setPatologieNom(?string $v): static { $this->patologieNom = $v; return $this; }
}
